### DISCUSSION TITLE
hope strongSwan support build-in vpn client on windows10

### DISCUSSION THREAD

**Discussion by ake047** [strongswan/strongswan#2539] [General]
I've deployed a VPN server using the latest version of strongSwan on CentOS 8, with EAP as the user authentication method. My goal is to allow my team members to connect to the VPN server using the built-in Windows 10 VPN client with username/password authentication, without requiring any additional configuration on their Windows 10 systems.

I understand that using the built-in Windows VPN client with the VPN type set to IKEv2 and entering a username and password results in a "policy match error" with error code 13868 in the Windows Event Viewer. While official workarounds exist, I don't believe they're universally applicable, as I can't expect my clients to implement such changes.

Therefore, my question is: how can I enable seamless connectivity between the built-in Windows 10 VPN client and my strongSwan VPN server without requiring any additional registry modifications or other manual configurations on the client side?

**Comment by tobiasbrunner** [comment]
> Therefore, my question is: how can I enable seamless connectivity between the built-in Windows 10 VPN client and my strongSwan VPN server without requiring any additional registry modifications or other manual configurations on the client side?

The only option is to add the `modp1024` DH group explicitly to the server's IKE proposal, which is **not recommended** due to [security concerns](https://docs.strongswan.org/docs/5.9/howtos/securityRecommendations.html#_weak_cryptographic_algorithms). Hence the [recommended configuration changes](https://docs.strongswan.org/docs/5.9/interop/windowsClients.html#strong_ke) on the client (via PowerShell, adding/updating the VPN connection could maybe be automated).

**Reply by ake047** [reply]
thank you very much for your answer.
